File: 0217-contains-duplicate/solution.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& v) {
        const int n = v.size() ;
        unordered_set<int > s ;
        for(auto i : v)
            s.insert(i);
        
        return n > s.size() ;
    } 
};

File: 0217-contains-duplicate/solution.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Map<Integer,Integer> mp = new HashMap<>();
        int n = nums.length ;

        for(int i = 0 ; i < n ; i++){
            if(mp.containsKey(nums[i])){
                mp.put(nums[i],mp.get(nums[i])+1);
            }
            else mp.put(nums[i],1);
        }

        for(Map.Entry<Integer,Integer> e : mp.entrySet()){
            if(e.getValue()>1)
            return true;
        }
        return false;
    }
}
